package.json

```
@@ -1,7 +1,7 @@
 {
   "name": "cuttle",
   "description": "The deepest card game under the sea",
-  "version": "4.4.4",
+  "version": "4.4.5",
```

Nit: this change would be a minor change, not a patch change

```suggestion
  "version": "4.5.0",
```

---

I think since this is just the backend update and there's no user-facing change yet that it's better represented as a patch. I'd consider the front end implementation a minor update when this data is actually visualized for users

---

If we're getting technical, changes to what data an API returns _are_ user-facing.  In the case of the API, the "user" is the frontend.

If there is something written to interface with the post-change version of the API, it might not function with the pre-change version of the API

According to [SemVer](https://semver.org/):

> Given a version number MAJOR.MINOR.PATCH, increment the:

> 1. MAJOR version when you make incompatible API changes
> 2. MINOR version when you add functionality in a backward compatible manner
> 3. PATCH version when you make backward compatible bug fixes

The changes you're proposing are backwards compatible (older consumers of the API won't need to change anything), but you're adding functionality and not just fixing bugs

---

Fair point; it's always helpful to come back to the documentation. That said, as I understand it, the 'by the book' definitions of SemVer don't really apply to applications in the first place, right? They were clearly written for tooling and dependencies. What does backwards compatible even mean for an application like Cuttle? Would it mean "being able to use an out-of-date client with the deployed API?"

That's a pretty weird use case (especially for the primary differentiator between version numbers), but it's the most coherent I can think of. Is that what you have in mind? So the rule would essentially be:

1. Every bugfix is a patch unless it requires changing the api in such a way that an out of date client... couldn't use the application? Would still experience the bug? In which case it becomes a major version change? I'm fuzzy on whether 'add functionality' is limited to new features or whether this also applies to bugs, but it sounds like every bugfix is either a patch or a major version from the above. Is that right?
2. Every new feature is a minor version change (including for example a single line of css, right?) unless it requires api changes in which case it's a major version? I'm still fuzzy on what would constitute 'incompatible' changes and whether I am right to hone in specifically on the idea of someone using a previous iteration of the frontend to connect directly to the current backend as the defining principle or if there is something else to use.

In part what's counterintuitive about this is that some bug fixes feel 'bigger' than some 'features' and visa versa. But I'd love arrive at a clearer and more consistent definition and guideline I can use for defining the version numbers, because I'll readily admit that I've been shooting from the hip. So this is helpful discourse! I think the key thing for me would be unpacking the notion of 'compatibility' in the context of applications, as well as wrapping my head around who the 'user' is given that some changes are dev-specific e.g. documentation, refactoring, linting, testging setup etc, and others are obviously player-centric. How do you frame your thinking about these issues @byarmis?

---

I’m with @byarmis — it’s completely possible (albeit very unlikely) that an old client would connect to the app. Unless we added an out of version check to the backend and had it return a 400 level error message, it’s still possible.

that said, I’m all for tightening up our own definition and while I do think there’s some level of guesswork we can certainly define and document our own ideas and rails.